invenio_drafts_resources/services/records/components/files.py

```diff
@@ -103,106 +93,51 @@ def update_draft(self, identity, data=None, record=None, errors=None):
+            errors.append(
+                {
+                    "field": f"f{self.files_data_key}.default_preview",
+                    "messages": e.messages,
+                }
+            )
 
     def edit(self, identity, draft=None, record=None):
         """Edit callback."""
-        if draft.bucket is None:
+        draft_files = self.get_record_files(draft)
+        record_files = self.get_record_files(record)
+        draft_bucket = self.get_record_bucket(draft)
+        if draft_bucket is None:
             # Happens, when a soft-deleted draft is un-deleted.
-            draft["files"] = {"enabled": True}
-            draft.files.create_bucket()
-        draft.files.copy(record.files)
-
-    def new_version(self, identity, draft=None, record=None):
-        """New version callback."""
-        # We don't copy files from the previous version, but instead allow
-        # users to import the files.
-        draft.files.enabled = record.files.enabled
-
-    def _publish_new(self, identity, draft, record):
-        """Action when publishing a new draft."""
-        # For unpublished drafts (new and new version), we move the draft
-        # bucket from the draft to the record (instead of creating a new, and
-        # deleting one). For consistency, we keep a bucket for all records
-        # independently of if they have files enabled or not.
-        record.files.set_bucket(draft.bucket)
-        record.files.copy(draft.files, copy_obj=False)
-
-        # Lock the bucket
-        # TODO: Make the locking step optional in the future (so
-        # instances can potentially allow files changes if desired).
-        record.files.lock()
-
-        # Cleanup
-        if draft.files.enabled:
-            draft.files.delete_all(remove_obj=False)
-        draft.files.unset_bucket()
+            draft[self.files_data_key] = {"enabled": record_files.enabled}
+            # re-fetch the files attribute - above data getter sets the attribute
+            draft_files = self.get_record_files(draft)
+            draft_files.create_bucket()
+        draft_files.copy(record_files)
+        # in the media files we don't lock the bucket
+        # - they can be simply edited from draft
 
     def _publish_edit(self, identity, draft, record):
-        """Action when publishing a edit to an existing record."""
+        """Action when publishing an edit to an existing record."""
         # TODO: For published records, we should sync changes from the
         # draft bucket to the record bucket, so that an instance could
         # potentially allow a user to update files. For now, sync() only
         # changes the default_preview and order
-        record.files.sync(draft.files)
+        record_files = self.get_record_files(record)
+        draft_files = self.get_record_files(draft)
+        record_files.sync(draft_files)
+        record_files.unlock()
+        record_files.copy(draft_files)
```

Comment:
media files are simply copied over from the draft - no new version required - see the RFC